snowshu/core/docker.py

```diff
@@ -144,20 +116,80 @@ def get_stopped_container(  # noqa pylint: disable=too-many-arguments
                         logger.info(
                             f'Base image is of target arch {arch}, using it...')
                         image = image_candidate
+                    elif len(arch_list) == 1:
+                        # If the build is not multiarch, we should still go through with it
+                        logger.warning(
+                            f'Base image is NOT of target arch {arch}, but only one arch was requested, continuing...')
+                        image = image_candidate
                     else:
                         # If supplied image is not of current arch, pull postgres instead
                         logger.info(
-                            f'Base image is NOT of target arch {arch}, using postgres instead...')
-
+                            f'Base image is NOT of target arch {arch}, using base db image instead...')
                         try:
                             image = self.client.images.get(
-                                f'{target_adapter.DOCKER_IMAGE.split(":")[0]}:{arch}')  # noqa pylint: disable=use-maxsplit-arg
+                                f'{target_adapter.BASE_DB_IMAGE.split(":")[0]}:{arch}')  # noqa pylint: disable=use-maxsplit-arg
                         except docker.errors.ImageNotFound:
                             image = self.client.images.pull(
-                                target_adapter.DOCKER_IMAGE, platform=f'linux/{arch}')
-                            image.tag(f'{target_adapter.DOCKER_IMAGE.split(":")[0]}:{arch}')  # noqa pylint: disable=use-maxsplit-arg
+                                target_adapter.BASE_DB_IMAGE, platform=f'linux/{arch}')
+                            image.tag(f'{target_adapter.BASE_DB_IMAGE.split(":")[0]}:{arch}')  # noqa pylint: disable=use-maxsplit-arg
+
+                except ConnectionError as error:
+                    logger.error(
+                        'Looks like docker is not started, please start docker daemon\nError: %s', error)
+                    raise
+
+                tagged_container_name = f'{name}_{arch}'
+                logger.info(
+                    f"Creating stopped container {tagged_container_name}...")
+                self.remove_container(tagged_container_name)
+
+                container = self.client.containers.create(
+                    image.tags[0],
+                    target_adapter.DOCKER_START_COMMAND,
+                    network=network.name,
+                    name=tagged_container_name,
+                    hostname=target_adapter._credentials.host,  # noqa pylint: disable=protected-access
```

Comment:
Can we not access `target_adapter.credentials` here? I thought that was wrapped with property decorators/functions.

Reply:
We can, it's just that underscored one caught my eye first

Reply:
Makes sense. Let's use the actual `credentials` attribute in this case then.